docs/development.md

```diff
@@ -0,0 +1,31 @@
+# Development Guide
+
+This document explains how to develop cluster-api-provide-openstack.
+
+## Using your own openstack-cluster-api-controller image for testing
+
+1. Building your own openstack-cluster-api-controller image
+1. Using your own openstack-cluster-api-controller image
+
+### Building your own openstack-cluster-api-controller image
+
+Set environment variables which is used in Makefile.
+
+Variable | Meaning
+------------ | -------------
+REGISTRY | The registy name 
+VERSION | The image version
+DOCKER_USERNAME | The username for logging in to the Docker registry
+DOCKER_PASSWORD | The password for logging in to the Docker registry
+
+Execute the command to build and upload the image to the Docker registry.
+
+
+```bash
+$ make upload-images
+
+```
+
+### Using your own openstack-cluster-api-controller image
+
+After generating `provider-components.yaml`, update `spec.template.spec.containers[].image` in the file according to the variables specified above.
```

Comment:
Which variable? Can you provide an example?

Reply:
We replace `k8scloudprovider` with REGISTRY and `latest` with VERSION respectively. I will add this.